libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -320,6 +320,14 @@ - (void)enqueueRequest:(SFRestRequest *)request delegate:(id<SFRestDelegate>)del
     }
 }
 
+- (SFNetwork *)networkForRequest:(SFRestRequest *)request {
```

Comment:
What happens if token refresh is thrown in the mix here? For example, let's assume we have 2 requests, A & B. A has been set up to use ephemeral session and B has been set up to use background session. Both of them are enqueued and 1 of them triggers the refresh flow.
- Which session is used for the refresh flow?
- Once refresh is done (assuming it's successful), when the requests are replayed (queue is flushed), do they retain their original session instances?

Reply:
@bhariharan All auth related calls should be using a dataTask &. non background sessions. SessionRefresher should be using its own(auth) related session. Once completion block is invoked the pendingRequests should get fired using the SFRestApi's shared session Right @bbirman?

Reply:
Yup! The sessions for auth and the pending requests would be separate-- @bhariharan let me know if I'm not understanding a piece